tests/Api/Parser/JsonParserTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+use Aws\Api\Parser\Exception\ParserException;
+use Aws\Api\Service;
+use Aws\AwsClient;
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+
+/*
+ * This class tests the PHP specific functionality of the JsonParser;
+ * cross-SDK testing is done in ComplianceTest.php
+ */
+
+/**
+ * @covers \Aws\Api\Parser\JsonRpcParser
+ * @covers \Aws\Api\Parser\JsonParser
+ */
+class JsonParserTest extends TestCase
```

Comment:
Should there also be tests for an operation which is modeled as a unix timestamp, and returns the same kinds of things tested for in these cases (i.e. returning a unix timestamp, text date, an iso8601 date, etc)?

Since many of the tests are essentially repeated code with different inputs and expectations, the PHPUnit [@dataProvider pattern ](https://phpunit.readthedocs.io/en/9.1/writing-tests-for-phpunit.html#data-providers)(used throughout the SDK) seems like it would be a good fit here.